### DISCUSSION TITLE
PyVista primitives in Gempy?

### DISCUSSION THREAD

**Discussion by Didymograptus** [gempy-project/gempy#778] [Q&A] [answered]
I've been getting to grips with the basics of modeling in Gempy, which is really awesome.  One thing I like to do when modeling is compare the surfaces to the data points, which Gempy can show, but my preferred way of doing this is to view the strata as shown in 3D rendered boreholes (cylinders), rather than just points.

Creating borehole 'sticks' from the data is no problem in PyVista and at the moment I export those tubes or cylinders to vtk or STL and view then in Paraview together with the model exported from Gempy ( as grid and/or surfaces.)  I've also been viewing the models in FreeCAD, which works well because Paraview doesn't seem very stable on my Win 10 PC.

This works fine, but it isn't a great workflow. Ideally I would like to be able to create the BH primitives (cylinders/tubes) in Gempy, but can't seem to find a way either to create them in the model or pull in external meshes.

Any help/guidance would be much appreciated.

**Comment by AlexanderJuestel** [comment, marked as answer]
@Didymograptus , 

great to see that you are using GemPy. Please have a look at our data processing package GemGIS (https://github.com/cgre-aachen/gemgis) which allows you to perform post processing tasks. Here is the link to a tutorial on how to export the meshes of GemPy as PyVista Meshes, you should then be able to add the wells as tubes to your PyVista plotter. 

https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/18_creating_depth_maps_from_gempy_models.html

Let me know if I can help you with anything else!
Alex

**Comment by Didymograptus** [comment]
Perfect, thanks. That is a great help. 

![image](https://user-images.githubusercontent.com/26096183/225956780-828adf79-a023-49e5-b11b-fdaad82b3916.png)
